pkg/controller/elasticsearch/validation/validations.go

```diff
@@ -33,6 +33,20 @@ var Validations = []Validation{
 	noBlacklistedSettings,
 	validSanIP,
 	pvcModification,
+	checkNodeSetNameUniqueness,
+}
+
+func checkNodeSetNameUniqueness(ctx Context) validation.Result {
+	nodeSets := ctx.Proposed.Elasticsearch.Spec.NodeSets
+	names := make(map[string]interface{})
+	for _, nodeSet := range nodeSets {
+		if _, found := names[nodeSet.Name]; found {
+			msg := fmt.Sprintf("duplicate node set '%s'", nodeSet.Name)
+			return validation.Result{Reason: fmt.Sprintf("%s: %s", validationFailedMsg, msg)}
```

Comment:
If there are two or more duplicates this loop will return early and only validate one by one. Not sure if it is of much practical relevance though